### PR TITLE
Refactor pullbacks

### PR DESCRIPTION
**Pull request by mortberg**
The definition of pullbacks was very deeply nested, so I flattened it. I also express the property of being a pullback square separately now as it's very useful to be able to talk about specific squares being pullbacks.

### CODE REVIEW

File: Cubical/Categories/Limits/Pullback.agda [hunk 1]
```diff
@@ -26,37 +28,41 @@ module _ {C : Category ℓ ℓ'} where
       s₁ : C [ l , m ]
       s₂ : C [ r , m ]
```
Comment (mortberg) [inline Cubical/Categories/Limits/Pullback.agda:30]: I'm not sure that it's worth having a separate record for cospans... What do you think @ecavallo ?

Comment (ecavallo) [reply Cubical/Categories/Limits/Pullback.agda:30]: I think it's fine to have. I think the category parameter should be explicit, though.

Comment (mortberg) [reply Cubical/Categories/Limits/Pullback.agda:30]: I wonder if it should be explicit to all definitions in the module or not... Hmm

Comment (mortberg) [reply Cubical/Categories/Limits/Pullback.agda:30]: I made it explicit

**Comment by mzeuner** [comment]
Is there anything you still want to add? If not, would you mind merging it, because I was planning on putting the new definitions to work for finite limits...

**Comment by mortberg** [comment]
> Is there anything you still want to add? If not, would you mind merging it, because I was planning on putting the new definitions to work...
> 
> Everything in this PR looks good to me btw.

Ah yes, I wanted to add an accessor function for the pullback arrow (the map one gets from the universal property) as well as its properties. I'll add this quickly now and then merge. Next thing on my list is to clean the code about initial/terminal objects

**Comment by mortberg** [comment]
Should be merged as soon as CI is done

**Pull request merged**